examples/ttn-otaa-feather-us915-dht22/ttn-otaa-feather-us915-dht22.ino

```diff
@@ -0,0 +1,248 @@
+/*******************************************************************************
+ * The Things Network - Sensor Data Example
+ * 
+ * Example of sending a valid LoRaWAN packet with DHT22 temperature and
+ * humidity data to The Things Networ using a Feather M0 LoRa.
+ * 
+ * Learn Guide: https://learn.adafruit.com/the-things-network-for-feather
+ * 
+ * Copyright (c) 2015 Thomas Telkamp and Matthijs Kooijman
+ * Copyright (c) 2018 Terry Moore, MCCI
+ * Copyright (c) 2018 Brent Rubell, Adafruit Industries
+ * 
+ * Permission is hereby granted, free of charge, to anyone
+ * obtaining a copy of this document and accompanying files,
+ * to do whatever they want with them without any restriction,
+ * including, but not limited to, copying, modification and redistribution.
+ * NO WARRANTY OF ANY KIND IS PROVIDED.
+ *******************************************************************************/
+#include <lmic.h>
+#include <hal/hal.h>
+#include <SPI.h>
+
+// include the DHT22 Sensor Library
+#include "DHT.h"
+
+// DHT digital pin and sensor type
+#define DHTPIN 10
+#define DHTTYPE DHT22
+
+// init. DHT
+DHT dht(DHTPIN, DHTTYPE);
+
+// This EUI must be in little-endian format, so least-significant-byte
+// first. When copying an EUI from ttnctl output, this means to reverse
+// the bytes. For TTN issued EUIs the last bytes should be 0xD5, 0xB3,
+// 0x70.
+static const u1_t PROGMEM APPEUI[8]= { 0xEF, 0x24, 0x01, 0xD0, 0x7E, 0xD5, 0xB3, 0x70 };
+void os_getArtEui (u1_t* buf) { memcpy_P(buf, APPEUI, 8);}
+
+// This should also be in little endian format, see above.
+static const u1_t PROGMEM DEVEUI[8]= { 0xD6, 0xDD, 0x10, 0x00, 0x00, 0xB6, 0x76, 0x98 };
+void os_getDevEui (u1_t* buf) { memcpy_P(buf, DEVEUI, 8);}
+
+// This key should be in big endian format (or, since it is not really a
+// number but a block of memory, endianness does not really apply). In
+// practice, a key taken from the TTN console can be copied as-is.
+static const u1_t PROGMEM APPKEY[16] = { 0xBE, 0xDE, 0xCE, 0x25, 0x06, 0x38, 0xD1, 0x4D, 0x8E, 0x73, 0xE9, 0xE5, 0x61, 0x9F, 0xCC, 0xC3 };
```

Comment:
There's a security issue here -- this is the shared secret between the device and the network. I think you should clone the `FILLMEIN` dodge from the feather example, for two reasons:

- fewer complaints from security mavens, and
- eliminate collisions when two people use the same sketch... the last one up will win.

I have heard rumors that TTN Europe has a way to dynamically assign these -- you use magic values here, and the first thing that happens after registration is that they send new messages with real credentials. But that will require more code and you might want to do that as the next step (possibly with the smarter `arduino-lorawan` wrapper).